log-collector/src/test/scala/io/phdata/pulse/logcollector/KafkaConsumerPulseTest.scala

```diff
@@ -0,0 +1,208 @@
+/* Copyright 2018 phData Inc. */
+
+package io.phdata.pulse.logcollector
+
+import org.scalatest.{ BeforeAndAfterEach, FunSuite }
+import java.util.Properties
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import spray.json._
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.solr.client.solrj.SolrQuery
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import io.phdata.pulse.logcollector.util.{ KafkaMiniCluster, ZooKafkaConfig }
+import io.phdata.pulse.testcommon.BaseSolrCloudTest
+import io.phdata.pulse.common.{ JsonSupport, SolrService }
+import io.phdata.pulse.common.domain.LogEvent
+
+class KafkaConsumerPulseTest
+    extends FunSuite
+    with BeforeAndAfterEach
+    with JsonSupport
+    with BaseSolrCloudTest {
+  // start kafka minicluster (broker)
+  var kafkaMiniCluster: KafkaMiniCluster = _
+  var zooKafkaConfig: ZooKafkaConfig     = _
+
+  implicit var actorSystem: ActorSystem             = _
+  implicit var actorMaterializer: ActorMaterializer = _
+
+  var solrService: SolrService            = _
+  var streamProcessor: KafkaConsumerPulse = _
+
+  override def beforeEach(): Unit = {
+    zooKafkaConfig = new ZooKafkaConfig
+    kafkaMiniCluster = new KafkaMiniCluster(zooKafkaConfig)
+    kafkaMiniCluster.start()
+
+    actorSystem = ActorSystem()
+    actorMaterializer = ActorMaterializer.create(actorSystem)
+
+    solrService = new SolrService(miniSolrCloudCluster.getZkServer.getZkAddress, solrClient)
+
+    streamProcessor = new KafkaConsumerPulse(solrService)
+  }
+
+  override def afterEach(): Unit = {
+    kafkaMiniCluster.stop()
+    solrService.close()
+  }
+
+  // Set topic name and messages
+  val TOPIC1 = "pulse_test"
+  val TOPIC2 = "solr_test"
+
+  // String JSON messages
+  val logMessage1: String =
```

Comment:
You could re-use an event generator like I discuss in the next comment, calling `toJson` on the messages, instead of hardcoding all these mock events.